src/wallet/migrate.cpp

```diff
@@ -557,6 +557,20 @@ void BerkeleyRODatabase::Open()
         throw std::runtime_error("BDB builtin encryption is not supported");
     }
 
+    // Check all LSNs point to file 0 and offset 1 which indicates that the LSNs were
+    // reset and that the log files are not necessary to get all of the data in the database.
+    for (uint32_t i = 0; i < outer_meta.last_page; ++i) {
+        // The LSN is composed of 2 32-bit ints, the first is a file id, the second an offset
+        // It will always be the first 8 bytes of a page, so we deserialize it directly for every page
+        uint32_t file;
+        uint32_t offset;
+        SeekToPage(db_file, i, page_size);
+        db_file >> file >> offset;
```

Comment:
The check currently always fails for other-endian wallets, needs something like:
```diff
diff --git a/src/wallet/migrate.cpp b/src/wallet/migrate.cpp
index 15478e8bc5bfe39e04b400fa475215e25eef6452..0a41ae199e1950bf372bae90b8a88fccca2b88f0 100644
--- a/src/wallet/migrate.cpp
+++ b/src/wallet/migrate.cpp
@@ -566,6 +566,10 @@ void BerkeleyRODatabase::Open()
         uint32_t offset;
         SeekToPage(db_file, i, page_size);
         db_file >> file >> offset;
+        if (outer_meta.other_endian) {
+            file = internal_bswap_32(file);
+            offset = internal_bswap_32(offset);
+        }
         if (file != 0 || offset != 1) {
             throw std::runtime_error("LSNs are not reset, this database is not completely flushed. Please reopen then close the database with a version that has BDB support");
         }
```

Reply:
Good catch, fixed